lib/puppet/file_serving/metadata.rb

```diff
@@ -136,7 +136,8 @@ def initialize(path,data={})
     @destination = data.delete('destination')
     @source      = data.delete('source')
     @content_uri = data.delete('content_uri')
-    super(path,data)
+
+    super(path, data.map { |k, v| [k.to_sym, v] }.to_h)
```

Comment:
Since this method gets called during metadata deserialization, what happens if an older agent (with this change) is talking to a newer puppetserver/fileserver (> 6.0) and the server returned additional fields in the metadata object that the agent didn't understand? Previously we would have raised "Unknown hash arguments" error. Ah it looks like ruby raises the same ArgumentError if the keyword is not known, so 👍 

```
irb(main):003:0> foo(b: 1)
ArgumentError: unknown keyword: b
```